baggage/baggage.go

```diff
@@ -230,6 +257,10 @@ type Member struct {
 // Notice: Consider using [NewMemberRaw] instead
 // that does not require percent-encoding of the value.
 func NewMember(key, value string, props ...Property) (Member, error) {
+	if !validateKey(key) {
```

Comment:
```suggestion
	if !validateBaggageName(key) {
```

Reply:
Update the comment to include the same information as for `NewMemberRaw`

Reply:
same as https://github.com/open-telemetry/opentelemetry-go/pull/5132#discussion_r1688954419